src/client/job.rs

```diff
@@ -57,20 +57,47 @@ pub fn job_status(info: &JobInfo) -> Status {
 fn task_status(status: &JobTaskState) -> Status {
     match status {
         JobTaskState::Waiting => Status::Waiting,
-        JobTaskState::Running => Status::Running,
-        JobTaskState::Finished => Status::Finished,
+        JobTaskState::Running(_) => Status::Running,
+        JobTaskState::Finished(_) => Status::Finished,
         JobTaskState::Failed(_) => Status::Failed,
         JobTaskState::Canceled => Status::Canceled,
     }
 }
 
-fn status_cell(status: Status) -> CellStruct {
+enum StatusColors {
+    Waiting,
+    Finished,
+    Failed,
+    Running,
+    Canceled,
+}
+
+fn get_color(status: StatusColors) -> Color {
     match status {
-        Status::Waiting => "WAITING".cell().foreground_color(Some(Color::Cyan)),
-        Status::Finished => "FINISHED".cell().foreground_color(Some(Color::Green)),
-        Status::Failed => "FAILED".cell().foreground_color(Some(Color::Red)),
-        Status::Running => "RUNNING".cell().foreground_color(Some(Color::Yellow)),
-        Status::Canceled => "CANCELED".cell().foreground_color(Some(Color::Magenta)),
+        StatusColors::Waiting => Color::Cyan,
+        StatusColors::Finished => Color::Green,
+        StatusColors::Failed => Color::Red,
+        StatusColors::Running => Color::Green,
+        StatusColors::Canceled => Color::Magenta
+    }
+}
+
+fn status_cell(status: Status, task_state: Option<&JobTaskState>) -> CellStruct {
+    match task_state {
+        Some(state) => match state {
+            JobTaskState::Waiting => "WAITING".cell().foreground_color(Some(get_color(StatusColors::Waiting))),
```

Comment:
* Status should be always only one of the keywords "RUNNING", "CANCELED", without any extra data (i.e. *not* RUNNING ON ...). 
Information about the worker should be another row/column in the table. There will be more information about worker in the future (e.g. CPU allocations) and we cannot have a super long status.

* If the task failed I also want to know on which worker the task was running.